### DISCUSSION TITLE
Why were authors removed?

### DISCUSSION THREAD

**Discussion by danielwilson1702** [android/nowinandroid#478] [Q&A] [answered]
Hi folks just wondering why authors were removed, seemed like a good idea to model authors as a common/relatable "item" in the app, as despite a large amount of code the app is relatively feature-light now.

Maybe there is some other plan to add more features, is there somewhere this kind of stuff is being discussed or is it all internal?

Thanks for the app I am learning a lot

**Comment by dturner** [comment, marked as answer]
Hi Daniel, 

Thanks for the question. We had quite a lot of discussion about whether to remove authors. Here's what we discussed: 

**Reasons to remove**

- Questionable user value - Users have told us that following an author is just a proxy for following topics, as in, authors generally only write about a few topics and it's simpler to just follow those topics directly.
- Takes up valuable space in the onboarding.
- Doesn't demonstrate much to developers which isn't already shown with Topics (see below).
- Additional workload for publishers. Each time a news resource is added which doesn't have an author, the author's bio and social links must be sourced. This adds overhead to an already cumbersome process.
- The current alphabetic ordering system doesn't work well, and to fix it requires work. Options:
  - Have a minimum threshold of articles (e.g. 3) which the author must publish before they are eligible to be followed
  - [Sort the author list by number of articles published](https://github.com/android/nowinandroid/pull/394) or some other criteria (popularity?)
 
**Reasons to keep**

- Adds personality to the app - it's not just about technical information, it's about the people who are delivering that information too. 
- Provides an example of the tabbed layout in the InterestsScreen.
- Provides an opportunity to show how to present variable string lengths ([example](https://github.com/android/nowinandroid/issues/408))

We have loads of ideas for other features, and this change will give us more bandwidth to focus on those. We'll try to share the roadmap as soon as we can.

**Reply by danielwilson1702** [reply]
Great OK thank you yes a roadmap would be super useful. I am basing a new project vaguely from the NIA structure and one of my fears is in 6 months time everyone decides the current way of doing things was a terrible idea, so when I saw large chunks of the code base disappear it raised my eyebrows 😂

**Comment by pulkit161001** [comment]
What is this nonsense !!!

**Reply by dturner** [reply]
Must've been a spam user. Looks like the content has been removed now.
